src/utils/auth/auth.provider.tsx

```diff
@@ -500,16 +505,22 @@ export function AuthRequestsProvider({ children }: PropsWithChildren) {
       } else if (!isEmbedded) {
         const isLocked = !(await getDecryptionKey());
 
+        // Check if we're within the initial request arrival window (AUTH_POPUP_REQUEST_ARRIVAL_WINDOW_MS ms after mounting)
+        // This gives time for pending auth requests to arrive before automatically closing the popup
+        const isInRequestArrivalWindow =
+          Date.now() - mountTimeRef.current <
+          AUTH_POPUP_REQUEST_ARRIVAL_WINDOW_MS;
+
         if (isLocked) {
           // If the wallet is locked, the user has `AUTH_POPUP_UNLOCK_REQUEST_TTL_MS` (15 minutes) to unlock it before
           // we close it automatically. While that's happening, AuthRequest might or might not be in this provide
           // already:
           clearCloseAuthPopupTimeout = closeAuthPopup(
             AUTH_POPUP_UNLOCK_REQUEST_TTL_MS
           );
-        } else if (!hasAuthRequests) {
+        } else if (!hasAuthRequests && !isInRequestArrivalWindow) {
           // Once the wallet is unlocked, we close the popup if an AuthRequest doesn't arrive in less than
-          // `AUTH_POPUP_REQUEST_WAIT_MS` (1 second):
+          // `AUTH_POPUP_REQUEST_WAIT_MS` (1 second) but only if the component is not in the initial request arrival window:
           clearCloseAuthPopupTimeout = closeAuthPopup(
             AUTH_POPUP_REQUEST_WAIT_MS
```

Comment:
I think the same result can be achieved by simply increasing `AUTH_POPUP_REQUEST_WAIT_MS` to 2 seconds, as this `useEffect` already runs when the component mounts.

Also, not sure this works as expected, because if the `useEffect` runs too early (so `isInRequestArrivalWindow = false`), then the popup won't close even if a request never arrives, because the `useEffect` won't re-run after `AUTH_POPUP_REQUEST_ARRIVAL_WINDOW_MS` has passed.

Reply:
The same issue occurs when updating AUTH_POPUP_REQUEST_WAIT_MS to 2 seconds also.